cli/internal/core/engine.go

```diff
@@ -272,19 +286,14 @@ func (e *Engine) generateTaskGraph(pkgs []string, taskNames []string, tasksOnly
 	return nil
 }
 
-// AddTask adds a task to the Engine so it can be looked up later.
-func (e *Engine) AddTask(task *Task) *Engine {
-	// If a root task is added, mark the task name as eligible for
-	// root execution. Otherwise, it will be skipped.
-	if util.IsPackageTask(task.Name) {
-		pkg, taskName := util.GetPackageTaskFromId(task.Name)
+// AddTask adds root tasks to the engine so they can be looked up later.
+func (e *Engine) AddTask(taskName string) {
+	if util.IsPackageTask(taskName) {
+		pkg, taskName := util.GetPackageTaskFromId(taskName)
 		if pkg == util.RootPkgName {
 			e.rootEnabledTasks.Add(taskName)
 		}
 	}
-
-	e.Tasks[task.Name] = task
```

Comment:
this was the "up front" store of known tasks. We resolve this lazily now as we go through packageTasks.